framework/include/constraints/ADMortarConstraint.h

```diff
@@ -36,6 +36,25 @@ class ADMortarConstraint : public MortarConstraintBase
    */
   void computeJacobian(Moose::MortarType mortar_type) override;
 
+#ifdef MOOSE_GLOBAL_AD_INDEXING
+
+  /**
+   * Get rid of AD derivative entries by dof index
+   */
+  void trimDerivative(const dof_id_type & remove_derivative_index, ADReal & var);
+
+  /**
+   * Get rid of interior node variable's derivatives
+   */
+  void
+  trimInteriorNodeDerivatives(const std::map<unsigned int, unsigned int> & primary_ip_lowerd_map,
+                              std::array<MooseVariable *, 3> & moose_var,
```

Comment:
There's kind of a vector physics assumption baked into this API. Could we change this to a `std::vector` in case someone ever wants to pass in a single nodal variable (or two or > 3)?

Reply:
Switched inputs to `std::vector`. Makes more sense now for general use.